### PR TITLE
PLNSRVCE-1098: provision and destroy hypershift by using hyperhisft cli

### PR DESCRIPTION
**Pull request by xinredhat**
1. Refresh installed and set up a new Hyperhsift cluster by using hypershift cli
2. CI `pipeline-service-test` connect to the new Hypershift to spawn OCP 4.12  
3. the CI load credentials from Bitwarden server
4. Removed `storageClassName` in PVC, so that it will pick the default storageclass
5. ci-runner image installs hypershift cli

Signed-off-by: Xin Jiang @xinredhat

### CODE REVIEW

File: ci/images/ci-runner/Dockerfile [hunk 1]
```diff
@@ -30,7 +41,8 @@ RUN set -x \
         xz-5.2.5 \
     && microdnf clean all
 COPY --from=grpc_cli /usr/local/bin/grpc_cli /usr/local/bin/
+COPY --from=hypershift /usr/local/bin/hypershift /usr/local/bin/
 COPY shared /tmp/image-build/shared
-RUN /tmp/image-build/shared/hack/install.sh --debug --bin argocd,go,jq,kubectl,tkn,yq \
+RUN /tmp/image-build/shared/hack/install.sh --debug --bin argocd,go,jq,kubectl,tkn,yq,bitwarden \
```
Comment (Roming22) [inline ci/images/ci-runner/Dockerfile:46]: please keep the iist ordered.

File: shared/hack/install.sh [hunk 2]
```diff
@@ -171,6 +174,21 @@ install_grpc_cli() {
     fi
 }
 
+install_hypershift() {
+    # Build from code
```
Comment (Roming22) [inline shared/hack/install.sh:178]: This should checkout a specific tag set in `shared/config/dependencies.sh`

Comment (xinredhat) [reply shared/hack/install.sh:178]: hyperhsift qe told me that we should take the latest one from main branch and I looked at the tag, the latest one  `2.0.0-20220406093220`  is too old.  I will put main as the version for now.

File: ci/images/ci-runner/Dockerfile (hunk 1 above)
Comment (xinredhat) [reply ci/images/ci-runner/Dockerfile]: fixed

File: shared/hack/install.sh (hunk 2 above)
Comment (Roming22) [reply shared/hack/install.sh:179]: Let's use `main` as the version number for the moment then.

**Comment by Roming22** [comment]
Don’t forget to squash the commits.

**Comment by prietyc123** [comment]
@xinredhat I am not sure if you are aware but we have something called as `fetch_bitwarden_secret` functionality in our code base already https://github.com/openshift-pipelines/pipeline-service/blob/main/operator/images/cluster-setup/content/bin/utils.sh#L86 

I was wondering if we can also use these for our use case 🤔

**Comment by xinredhat** [comment]
> @xinredhat I am not sure if you are aware but we have something called as `fetch_bitwarden_secret` functionality in our code base already https://github.com/openshift-pipelines/pipeline-service/blob/main/operator/images/cluster-setup/content/bin/utils.sh#L86
> 
> I was wondering if we can also use these for our use case 🤔

yes, but this part of logic is different. If possible, we can refactor our code to avoid the duplicate code

**Review by Roming22 — APPROVED** [review]

**Pull request merged**